joskalenda/blog-app
# Validation n+1 and modelspec

# Blog App: validations, model specs, and n+1 problems ✅ 

## Learning objectives
- Use validations for models.
- Write unit tests for models.
- Secure app from n+1 problems.

- [ ] For the User model: ✅ 

1. Name must not be blank.
2. PostsCounter must be an integer greater than or equal to zero.

- [ ] For the Post model: ✅ 

1. Title must not be blank.
2. Title must not exceed 250 characters.
3. CommentsCounter must be an integer greater than or equal to zero.
4. LikesCounter must be an integer greater than or equal to zero.

- Add unit specs for all of your models' methods and validations. ✅ 
> | |Test case| |
> |-|-------|-| 
> | | <img width="531" alt="test" src="https://user-images.githubusercontent.com/77843050/167256082-73fe4f20-98a0-474e-b853-8d003d025bda.png"> | |

- Add flash messages in the create actions in all your controllers. ✅ 
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user ✅
